passes/src/Luna/Pass/Resolution/DeconstructorResolution.hs

```diff
@@ -15,50 +19,70 @@ import           Data.Maybe        (isJust)
+type instance Inputs     Layer DeconstructorResolution = '[AnyExpr // Model, AnyExpr // Type, AnyExpr // Succs, AnyExpr // UID, AnyExpr // Requester, AnyExpr // Errors, AnyExprLink // Model, AnyExprLink // UID]
 type instance Inputs     Attr  DeconstructorResolution = '[NegativeConses, Imports, Unifications]
 type instance Inputs     Event DeconstructorResolution = '[]
 
 type instance Outputs    Net   DeconstructorResolution = '[AnyExpr, AnyExprLink]
-type instance Outputs    Layer DeconstructorResolution = '[AnyExpr // Model, AnyExpr // Type, AnyExpr // Succs, AnyExpr // UID, AnyExprLink // Model, AnyExprLink // UID]
+type instance Outputs    Layer DeconstructorResolution = '[AnyExpr // Model, AnyExpr // Type, AnyExpr // Succs, AnyExpr // UID, AnyExpr // Requester, AnyExpr // Errors, AnyExprLink // Model, AnyExprLink // UID]
 type instance Outputs    Attr  DeconstructorResolution = '[NegativeConses, Unifications]
 type instance Outputs    Event DeconstructorResolution = '[New // AnyExpr, New // AnyExprLink, Event.Import // AnyExpr, Event.Import // AnyExprLink, Delete // AnyExpr, Delete // AnyExprLink, OnDeepDelete // AnyExpr]
 
 type instance Preserves        DeconstructorResolution = '[]
 
+instance MonadGetter s m => MonadGetter s (ExceptT r m)
+instance MonadSetter s m => MonadSetter s (ExceptT r m)
+
 runDeconstructorResolution :: (MonadRef m, MonadPassManager m) => Pass DeconstructorResolution m
 runDeconstructorResolution = do
     conses <- unwrap <$> getAttr @NegativeConses
-    mapM_ resolveDecons conses
+    forM_ conses $ \cons -> do
+        res <- resolveDecons cons
+        case res of
+            Left err -> modifyLayer_ @Errors cons (CompileError err [] [] :)
+            _        -> return ()
     putAttr @NegativeConses $ NegativeConses []
 
+wrongArgumentsCountError :: Int -> Int -> Name -> Text
+wrongArgumentsCountError expected given cons =  "Wrong number of arguments for constructor "
+                                             <> fromString (show cons)
+                                             <> ". Expected "
+                                             <> convert (show expected)
+                                             <> " but got "
+                                             <> convert (show given)
+                                             <> "."
+
 lookupDestr :: Name -> Imports -> Either ImportError Destructor
 lookupDestr n imps = case itoListOf (importedClasses .> itraversed <. Function.documentedItem . Class.constructors . ix n) imps of
     []       -> Left SymbolNotFound
     [(_, d)] -> Right $ snd d
     matches  -> Left . SymbolAmbiguous $ fst <$> matches
 
-resolveDecons :: (MonadRef m, MonadPassManager m) => Expr Cons -> SubPass DeconstructorResolution m (Either ImportError ())
+resolveDecons :: (MonadRef m, MonadPassManager m) => Expr Cons
```

Comment:
Is there a reason this isn't `resolveDeconstructor`? I don't find `resolveDecons` to be a hugely clear name.